### DISCUSSION TITLE
Obsidian like Graph view

### DISCUSSION THREAD

**Discussion by JustYannicc** [AppFlowy-IO/AppFlowy#171] [Ideas]
The only reason I use obsidian is because of that graph view it is really great and If you could implement it that would be cool

**Comment by lawweiliang** [comment]
@JustYannicc 

I did using Obsidian but I didn't not find the reason behind having Graph View aside from a "Cool feature" . Would you mind to share with me your use case?

**Reply by crossXproduct** [reply]
Hi @lawweiliang...so I'm not sure about @JustYannicc, but thought I'd take a stab at this question since I'm currently looking for a Notion alternative with this feature. I'm a researcher, and for me the graph view is a helpful way of organizing ideas and making connections between them. The standard hierarchical views are too rigid to support the way I think and work. Obsidian, on the other hand, is not structured enough...I need a database framework to keep my files from spiraling into utter chaos. But it would be extremely helpful to be able to see everything from a bird's-eye view, as my mind easily gets lost in the endless hierarchies.
Hope this helps!

**Reply by lawweiliang** [reply]
Hi @crossXproduct, thanks for your explanation.

 I think it would be better for you to draw out the graph view that you are expecting from the graph view.
 It would be better for us to grasp your thinking. 
 
 Oh Wait, you just mentioned that this is the feature of the notion. Would you mind print screen that and let us take a look?

**Reply by crossXproduct** [reply]
Sure thing...it's actually an Obsidian feature, not a Notion one, though a few people have tried implementing workarounds. Here is an Obsidian implementation: https://help.obsidian.md/Plugins/Graph+view
<img width="1383" alt="Screen Shot 2023-03-01 at 12 27 41 PM" src="https://user-images.githubusercontent.com/29285370/222216442-75f971bf-d58a-44dd-b4b1-6a06bb2eabb3.png">
and here is a wrapper type thing someone made for Notion (though I haven't been able to get it to work for me): https://lapis-aletopelta-fd7.notion.site/Entry-e5cae7e20c4248f989107715e565554b
<img width="1379" alt="Screen Shot 2023-03-01 at 12 27 54 PM" src="https://user-images.githubusercontent.com/29285370/222216457-04e1c720-1d36-4036-bb19-aab1c9d2160e.png">
The graph is interactive, so you can move the nodes around and open pages by clicking them.

If you work with text/markdown all the time and are comfortable with it, this is probably just a novelty, as you say. It's mainly for the more visually-oriented among us :).

**Reply by annieappflowy** [reply]
This is an example of my OB graph:
<img width="390" alt="image" src="https://user-images.githubusercontent.com/12026239/222314952-1afaad8d-0001-49f7-b003-331d52559347.png">

**Reply by bitsondatadev** [reply]
> Obsidian's graph view is a helpful way of organizing ideas and making connections between them. Notion's hierarchical views are too rigid to support abstract and creative reasoning. While I do need a database framework to keep things organized, the graph view lets me navigate connections and patterns over the endless hierarchies.

I've paraphrased @crossXproduct as it describes my use case for both Notion databases and Graph view as well. A really good example is from my friend @sspaeti who shares portions of his [second brain in obsidian](https://www.ssp.sh/brain/eltp/) with the community. Notice the interactive graph below as you read through each concept. This is incredibly valuable for sharing knowledge as well, be it with an entire community or just within teams.

Needless to say, huge +1 here!

From a product perspective, this would solve a large overlap problem in both Obsidian and Notion communities on top of you all offering a private Notion option. Don't know how long this will be a differentiating feature for AppFlowy but IMO it's a solid awareness play for AppFlowy in Notion and Obsidian communities (See: [1](https://www.reddit.com/r/ObsidianMD/comments/16ncqct/has_anyone_found_a_good_way_to_replicate_notions/), [2](https://forum.obsidian.md/t/replicate-notions-databases-with-relation-properties/52011/5), [3](https://www.reddit.com/r/Notion/comments/owu4ao/obsidians_graph_view_in_notion/)).

![Screenshot from 2024-07-03 06-29-40](https://github.com/AppFlowy-IO/AppFlowy/assets/8547669/46872a2e-c9ec-482b-b185-41d26026a2f2)

**Reply by bitsondatadev** [reply]
After typing all this out, it seems like [anytype covers all these bases](https://doc.anytype.io/anytype-docs/basics/graph), but is still pretty early.

**Reply by sspaeti** [reply]
I'm not sure about the use-case, but in case it's web based, I'd definitely take a look at https://github.com/jackyzha0/quartz. It is what I use for publishing my Obsidian Graph as a Website. Jacky, the creator, did a lot with [v3 with Hugo](https://github.com/jackyzha0/quartz/tree/hugo) (this is the version I use), but also with the latest version, where he re-wrote the whole thing. So I'm guessing, you'd find all you need in that repo, in case you want a web solution.

**Comment by Fau57** [comment]
Could probably use anytype or affines graph stuff in replacement. 
Just a thought.

**Comment by PNDTG** [comment]
It's not a feature everyone would use on the daily but it can be very helpful to quickly visualize all your knowledge at once, for quick access or brainstorming ideas.
It would be just as awesome if there was a plugin that did this, or could be an extra feature you could access and contribute to.

**Comment by MaleicAcid** [comment]
+1 really need this

**Comment by Moshe-Shelomov** [comment]
I would love to see that feature implemented as well
Also I thought it might be useful to point out that there is an open source program that has graph-centered workflow, its called "logseq"
I though it might be useful to tell about it, so people can look at their code or something, in case it will be decided to implement this feature in AppFlowy

**Comment by saxcorp** [comment]
+1 really need this too ^^
